app/pulp/app/viewsets/user.py

```diff
@@ -0,0 +1,23 @@
+from pulp.app.models import PulpUser
+from pulp.app.serializers import UserSerializer
+from pulp.app.viewsets import NamedModelViewSet
+
+from rest_framework import decorators, status
+from rest_framework.response import Response
+
+class UserViewSet(NamedModelViewSet):
+    endpoint_name = 'users'
+    queryset = PulpUser.objects.all()
+    serializer_class = UserSerializer
+
+    @decorators.detail_route(methods=['post'])
+    def set_password(self, request, pk=None):
+        user = self.get_object()
+        serializer = UserSerializer(data=request.data)
```

Comment:
The request data should contain only {'password': 'new_password'} right? I am curious if this would fail if any of the fields on the serializer other than password are changed to `required=True`.